pkg/apis/azure/validation/infrastructure.go

```diff
@@ -105,13 +107,19 @@ func ValidateInfrastructureConfig(infra *apisazure.InfrastructureConfig, shoot *
 	// validation here will fail for those cases.
 	// TODO: remove the following block and uncomment below blocks once deployment into existing resource groups works properly.
 	if infra.ResourceGroup != nil {
+		allErrs = append(allErrs, validateResourceGroup(infra.ResourceGroup.Name, fldPath.Child("resourceGroup.name"))...)
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("resourceGroup"), infra.ResourceGroup, "specifying an existing resource group is not supported yet"))
 	}
 
 	allErrs = append(allErrs, validateNetworkConfig(shoot, infra, nodes, pods, services, fldPath)...)
 
-	if infra.Identity != nil && (infra.Identity.Name == "" || infra.Identity.ResourceGroup == "") {
-		allErrs = append(allErrs, field.Invalid(fldPath.Child("identity"), infra.Identity, "specifying an identity requires the name of the identity and the resource group which hosts the identity"))
+	if infra.Identity != nil {
+		path := fldPath.Child("identity")
+		if infra.Identity.Name == "" || infra.Identity.ResourceGroup == "" {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("identity"), infra.Identity, "specifying an identity requires the name of the identity and the resource group which hosts the identity"))
```

Comment:
nit: maybe separate error messages for `Name` and `ResourceGroup` to be a bit more specific.